use-cases/virgo/data.py

```diff
@@ -61,16 +66,154 @@ def execute(self) -> pd.DataFrame:
+            ) for i in range(df_aux_all_2d.shape[0])]
+        )
+
+        # Concatenate the main and auxiliary channel tensors
+        data_tensor = torch.cat([signal_data_train_2d, aux_data_train_2d], dim=1)
+
+        # Normalize and return the concatenated tensor
+        return normalize_(data_tensor)
+
+
 class TimeSeriesDatasetSplitter(DataSplitter):
     def __init__(
         self,
         train_proportion: int | float,
         validation_proportion: int | float = 0.0,
         test_proportion: int | float = 0.0,
-        rnd_seed: int | None = None,
-        images_dataset: str = "data/Image_dataset_synthetic_64x64.pkl",
-        name: str | None = None
+        rnd_seed: Optional[int] = None,
+        root_folder: Optional[str] = None,
+        name: Optional[str] = None
     ) -> None:
+        """Initialize the splitter for time-series datasets.
+
+        Args:
+            train_proportion (int | float): Proportion of files for the training set.
+            validation_proportion (int | float): Proportion for validation.
+            test_proportion (int | float): Proportion for testing.
+            rnd_seed (int | None): Seed for randomization (optional).
+            root_folder (str | None): Folder containing the dataset files.
+            name (str | None): Name of the data splitter.
+        """
+        super().__init__(
+            train_proportion,
+            validation_proportion,
+            test_proportion,
+            name
+        )
+        self.save_parameters(**self.locals2params(locals()))
+        self.rnd_seed = rnd_seed
+        self.root_folder = root_folder
+
+    @monitor_exec
+    def execute(self) -> Tuple[Dataset, Dataset, Dataset]:
+        """Execute the dataset splitting process.
+
+        Finds all pickled files in the root folder, then splits them into
+        training, validation, and test sets based on the specified proportions.
+
+        Returns:
+            Tuple[Dataset, Dataset, Dataset]: Training, validation, and test datasets.
+        """
+
+        # Ensure the root folder exists
+        if not os.path.isdir(self.root_folder):
+            raise FileNotFoundError(f"Root folder '{self.root_folder}' not found.")
+
+        # Find all file paths in root folder
+        all_file_paths = [os.path.join(dirpath, f)
+                          for dirpath, dirs, files in os.walk(self.root_folder)
+                          for f in files if f.endswith('.pkl')]
+
+        # Split file paths into train, validation, and test sets
+        [train_paths, validation_paths, test_paths] = random_split(
+            all_file_paths,
+            [self.train_proportion,
+             self.validation_proportion,
+             self.test_proportion]
+        )
+
+        # Create dataset objects for each split
+        train_dataset = DataFrameDataset(train_paths)
+        validation_dataset = DataFrameDataset(validation_paths)
+        test_dataset = DataFrameDataset(test_paths)
```

Comment:
Consider creating a single `DataFrameDataset` on the whole dataset and then splitting it into train, test, val using https://pytorch.org/docs/stable/data.html#torch.utils.data.random_split

It should work as well and it would simplify your code:

1. Move the recursive exploration of the filesystem (looking for `pkl` files) in the constructor of `DataFrameDataset`.
2. Split the big dataset using `torch.utils.data.random_split` (preferably you need to pass a generator with a seed)

Reply:
It *should* work